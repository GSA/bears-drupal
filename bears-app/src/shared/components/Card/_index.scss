.relative-benefit-card {
  // .usa-card:last-child overide
  margin-bottom: 16px !important;
  /* stylelint-disable selector-class-pattern */
  .usa-card__container {
    background-color: #d9e8f6;
    border-radius: 4px;
    color: #162e51;
    height: auto;
    margin: 0;
  }

  a {
    text-decoration: none;
  }

  a:hover .usa-card__container {
    background-color: #b6dbfd;
  }

  a:active .usa-card__container {
    border: 2px solid #005ea2;
  }

  .usa-card__header {
    padding-bottom: 0;
  }

  .usa-card__body {
    padding-top: 0;
    padding-bottom: 24px;
    margin-top: 0;
    margin-bottom: 0;
  }
  /* stylelint-enable selector-class-pattern */
}
