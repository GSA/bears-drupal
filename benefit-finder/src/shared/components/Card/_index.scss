@use '../../styles/colors/_index.scss' as color;
@use '../../styles/space/_index.scss' as space;
@use '../../styles/mixins/_index.scss' as *;

#benefit-finder {
  .relative-benefit-card {
    // .usa-card:last-child overide
    margin-bottom: space.$space-md !important;

    .usa-card__container {
      background-color: color.$sky;
      border-radius: 4px;
      color: color.$officer-navy;
      height: auto;
      margin: 0;
    }

    a {
      text-decoration: none;
    }

    a:hover .usa-card__container {
      background-color: color.$dark-sky;
    }

    a:active .usa-card__container {
      border: 2px solid color.$pop-blue;
    }

    .usa-card__header {
      padding-bottom: 0;
    }

    h3.usa-card__heading {
      @include h6;
    }

    .usa-card__body {
      @include p;

      padding-top: 0;
      padding-bottom: space.$padding-bottom-md-plus;
      margin-top: 0;
      margin-bottom: 0;
    }
  }
}
