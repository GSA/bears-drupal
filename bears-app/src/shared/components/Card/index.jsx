import PropTypes from 'prop-types'
import { Heading, Carrot, Paragraph } from '../index'
import { useHandleClassName } from '../../hooks/useHandleClassName'
import './_index.scss'

/**
 * a functional component that renders a usa-card component
 * @component
 * @param {string} className - inherited class(es)
 * @param {string} href - location
 * @param {string} title - content
 * @param {string} body - content
 * @param {noCarrot} boolean - adds a decorative carrot to the link
 * @return {html} returns a semantic html list element
 */
const Card = ({ className, title, body, href, noCarrot }) => {
  const defaultClasses = ['usa-card add-list-reset']
  return (
    <li
      className={useHandleClassName({
        className,
        defaultClasses,
      })}
    >
      <a href={href}>
        <div className="usa-card__container">
          <div className="usa-card__header">
            <Heading className="usa-card__heading" headingLevel={3}>
              {title}
            </Heading>
          </div>
          <Paragraph className="usa-card__body">{body}</Paragraph>
          {noCarrot === true ? null : <Carrot color="#162E51" />}
        </div>
      </a>
    </li>
  )
}

Card.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  body: PropTypes.string,
  href: PropTypes.string,
  noCarrot: PropTypes.bool,
}

export default Card
