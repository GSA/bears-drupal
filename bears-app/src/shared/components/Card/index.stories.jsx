import Card from './index.jsx'

export default {
  component: Card,
  tags: ['autodocs'],
  args: {
    className: 'relative-benefit-card',
    title: 'Header Content',
    cta: 'BodyContent',
    link: 'https://www.google.com',
  },
}

export const Primary = {}
