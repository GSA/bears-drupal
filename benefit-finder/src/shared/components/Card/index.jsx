import PropTypes from 'prop-types'
import { Heading, Carrot } from '../index'
import { useHandleClassName } from '../../hooks/useHandleClassName'
import createMarkup from '../../utils/createMarkup'
import './_index.scss'

/**
 * a functional component that renders a usa-card component
 * @component
 * @param {string} className - inherited class(es)
 * @param {string} href - location
 * @param {string} title - content
 * @param {string} body - content
 * @param {string} cta - content
 * @param {noCarrot} boolean - adds a decorative carrot to the link
 * @return {html} returns a semantic html list element
 */
const Card = ({ className, title, body, cta, href, noCarrot, carrotType }) => {
  const defaultClasses = ['usa-card add-list-reset']
  const handleCarrot =
    noCarrot === true ? null : <Carrot color="#162E51" type={carrotType} />

  return (
    <li
      className={useHandleClassName({
        className,
        defaultClasses,
      })}
      key={`${title}`}
    >
      <a href={href}>
        <div className="usa-card__container">
          <div className="usa-card__header">
            <Heading className="usa-card__heading" headingLevel={3}>
              {title}
            </Heading>
          </div>
          <div
            className="usa-card__body"
            dangerouslySetInnerHTML={createMarkup(body)}
          />
          <div className="usa-card__cta">{cta}</div>
          {handleCarrot}
        </div>
      </a>
    </li>
  )
}

Card.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  body: PropTypes.string,
  href: PropTypes.string,
  noCarrot: PropTypes.bool,
}

export default Card
