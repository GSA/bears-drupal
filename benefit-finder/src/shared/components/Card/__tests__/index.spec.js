import renderer from 'react-test-renderer'
import Card from '../index.jsx'

describe('Card', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<Card />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
