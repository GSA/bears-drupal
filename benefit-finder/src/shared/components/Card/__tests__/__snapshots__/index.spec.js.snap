// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders a match to the previous snapshot 1`] = `
<li
  className=""
>
  <a>
    <div
      className="usa-card__container"
    >
      <div
        className="usa-card__header"
      >
        <h3
          className=""
          id=""
        />
      </div>
      <div
        className="usa-card__body"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
      />
      <div
        className="usa-card__cta"
      />
      <svg
        className="carrot"
        fill="#162E51"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </a>
</li>
`;
